### ISSUE TITLE
Allow feeding Pandas DataFrames directly to CSV datasets

### ISSUE THREAD

**Issue by holgerroth** [Project-MONAI/MONAI#3430]
**Is your feature request related to a problem? Please describe.**
In some situations, the user has not access to the csv files but some already generated pandas objects only. 

**Describe the solution you'd like**
In that case, it would be useful to create `CSVDataset` and `CSVIterableDataset` directly from the pandas DataFrames.

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

**Comment by Nic-Ma** [comment]
Hi @holgerroth ,

Thanks for your feature request.
It makes sense to me, I will try to enhance the `CSVDataset` and `CSVIterableDataset` for it soon.

Thanks.

**Comment by Nic-Ma** [comment]
Hi @holgerroth ,

BTW, I think `CSVIterableDataset` is designed to partly load extremely big CSV file or streaming, if you already have the DataFrame in memory, I think no need to use this class, so I will not enhance it.
What do you think?

Thanks.

**Comment by holgerroth** [comment]
Good point. I still think that the current class only accepts a filename as input and tries to use [pd.read_csv()](https://github.com/Project-MONAI/MONAI/blob/3960a514a5f44a10aceab463e926a19bc0b8c79a/monai/data/iterable_dataset.py#L211) from disk. It might be useful if it can accept the streaming object directly as well.

**Comment by Nic-Ma** [comment]
Hi @holgerroth ,

OK, it makes sense to accept streaming iterable objects here:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/iterable_dataset.py#L211

Thanks.
